tests/framework/utils.py

```diff
@@ -190,3 +191,41 @@ def run_cmd(cmd, ignore_return_code=False, no_shell=False):
         run_cmd_async(cmd=cmd,
                       ignore_return_code=ignore_return_code,
                       no_shell=no_shell))
+
+
+def get_threads(pid):
+    """
+    Return dict consisting of child threads.
+
+    {
+        "name": thread_pid
+    }
+    """
+    
+    ret_val = {}
+
+    proc = psutil.Process(pid)
+    for thread in proc.threads():
+        ret_val[psutil.Process(thread.id).name()] = thread.id
```

Comment:
AFAIK the thread name is not unique. So this is error prone.

Reply:
Mapped each thread name to a list. This will catch the threads with the same name.